textx/metamodel.py

```diff
@@ -140,6 +151,9 @@ def __init__(self, file_name=None, classes=None, builtins=None,
         # Registered object processors
         self.obj_processors = {}
 
+        # Registered scope provider
+        self.scope_provider = {}
```

Comment:
I'm a bit nitpicking here but `scope_provider` -> `scope_providers`.

Reply:
Quality starts with proper names! You are absolutely right. I fixed this inconsistency.